src/main/java/org/kohsuke/github/GHWorkflow.java

```diff
@@ -122,8 +122,14 @@ public void dispatch(String ref) throws IOException {
      *            default properties configured in the workflow file will be used when inputs are omitted.
      * @throws IOException
      *             the io exception
+     * @throws IllegalArgumentException
+     *             the illegal argument exception
      */
-    public void dispatch(String ref, Map<String, Object> inputs) throws IOException {
+    public void dispatch(String ref, Map<String, Object> inputs) throws IOException, IllegalArgumentException {
+        if (ref == null || ref.length() == 0) {
+            throw new IllegalArgumentException("ref cannot be null or empty");
+        }
+
```

Comment:
I think you'll also need to update Line 142 to use `send()`. 

What happens if `ref` is an invalid/missing ref?

Reply:
Yes, that makes sense. Thank you for spotting it! Fixed.